libraries/AP_Camera/AP_Camera_SoloGimbal.cpp

```diff
@@ -0,0 +1,89 @@
+#include "AP_Camera_SoloGimbal.h"
+#include <GCS_MAVLink/GCS.h>
+
+uint8_t AP_Camera_SoloGimbal::gopro_capture_mode;
+uint8_t AP_Camera_SoloGimbal::gopro_status;
```

Comment:
```suggestion
GOPRO_HEARTBEAT_STATUS AP_Camera_SoloGimbal::gopro_status;
```

Reply:
Didn't work

Reply:
```
diff --git a/libraries/AP_Camera/AP_Camera_SoloGimbal.cpp b/libraries/AP_Camera/AP_Camera_SoloGimbal.cpp
index cface37cc1..f86ca53455 100644
--- a/libraries/AP_Camera/AP_Camera_SoloGimbal.cpp
+++ b/libraries/AP_Camera/AP_Camera_SoloGimbal.cpp
@@ -1,8 +1,8 @@
 #include "AP_Camera_SoloGimbal.h"
 #include <GCS_MAVLink/GCS.h>
 
-uint8_t AP_Camera_SoloGimbal::gopro_capture_mode;
-uint8_t AP_Camera_SoloGimbal::gopro_status;
+GOPRO_CAPTURE_MODE AP_Camera_SoloGimbal::gopro_capture_mode;
+GOPRO_HEARTBEAT_STATUS AP_Camera_SoloGimbal::gopro_status;
 bool AP_Camera_SoloGimbal::gopro_is_recording;
 mavlink_channel_t AP_Camera_SoloGimbal::heartbeat_channel;
 
@@ -80,8 +80,8 @@ void AP_Camera_SoloGimbal::handle_gopro_heartbeat(mavlink_channel_t chan, const
 {
     mavlink_gopro_heartbeat_t report_msg;
     mavlink_msg_gopro_heartbeat_decode(&msg, &report_msg);
-    gopro_status = report_msg.status;
-    gopro_capture_mode = report_msg.capture_mode;
+    gopro_status = (GOPRO_HEARTBEAT_STATUS)report_msg.status;
+    gopro_capture_mode = (GOPRO_CAPTURE_MODE)report_msg.capture_mode;
     gopro_is_recording = report_msg.flags & GOPRO_FLAG_RECORDING;
     heartbeat_channel = chan;
 }
diff --git a/libraries/AP_Camera/AP_Camera_SoloGimbal.h b/libraries/AP_Camera/AP_Camera_SoloGimbal.h
index 87cf761eaa..08ffdb1902 100644
--- a/libraries/AP_Camera/AP_Camera_SoloGimbal.h
+++ b/libraries/AP_Camera/AP_Camera_SoloGimbal.h
@@ -11,8 +11,8 @@ public:
 
 private:
 
-    static uint8_t gopro_capture_mode;
-    static uint8_t gopro_status;
+    static GOPRO_CAPTURE_MODE gopro_capture_mode;
+    static GOPRO_HEARTBEAT_STATUS gopro_status;
     static bool gopro_is_recording;
     static mavlink_channel_t heartbeat_channel;
 };
```

('though see my mumblings on validation rather than the straight-cast you see there)